report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,357 @@
+    @observable private selectedOptionId;
+    private _methodNameInput;
+
+    constructor(
+        private _statusConverter: StatusConverter,
+        private _statisticsGenerator: StatisticsGenerator,
+    ) {
+        super();
+        this._bars = [];
+        this._lookUpOptions = [];
+        this._option = {};
+    }
+
+    activate(params: any, routeConfig: RouteConfig, navInstruction: NavigationInstruction) {
+        super.activate(params, routeConfig, navInstruction);
+        if (!this.queryParams.rangeNum) this.queryParams.rangeNum = '10';// only set range 10 if there is no range at all (e.g. when navigation from another view)
+        if (params.config) {
+            this._showConfigurationMethods = !!params.config.toLowerCase();
+        }
+        this._getLookUpOptions()
+        this._filter();
+    }
+
+    selectedOptionIdChanged(methodId: string) {
+        if (methodId) {
+            this._methodId = methodId;
+            this.queryParams.methodId = methodId
+            this.updateUrl(this.queryParams);
+            this._highlightData()
+        }
+    }
+
+    selectionChanged() { // called when input from lookup text field is deleted manually
+        this._setChartOption(); // overwrites color highlighting (reset)
+        if (this._inputValue.length == 0) {
+            this._methodId = undefined;
+            delete this.queryParams.methodId;
+            this.updateUrl({rangeNum: this.queryParams.rangeNum, methodId: this._methodId});
+            this._getLookUpOptions();
+        }
+    }
+
+    private _getLookUpOptions() {
+        this._statisticsGenerator.getExecutionStatistics().then(executionStatistics => {
+            let methodContexts: IMethodContext[];
+            if (this.queryParams.methodId) {
+                methodContexts = [executionStatistics.executionAggregate.methodContexts[this.queryParams.methodId]];
+                this._highlightData();
+                this.updateUrl({methodId: this.queryParams.methodId});
+                this._methodId = this.queryParams.methodId;
+            } else if (this._inputValue?.length > 0) {
+                this._searchRegexp = this._statusConverter.createRegexpFromSearchString(this._inputValue);
+                delete this.queryParams.methodId;
+                methodContexts = Object.values(executionStatistics.executionAggregate.methodContexts)
+                    .filter(methodContext => methodContext.contextValues.name.match(this._searchRegexp))
+            } else {
+                methodContexts = Object.values(executionStatistics.executionAggregate.methodContexts);
+            }
+            if (!this._showConfigurationMethods) {
+                methodContexts = methodContexts.filter(methodContext => methodContext.methodType == MethodType.TEST_METHOD);
+            }
+            this._lookUpOptions = methodContexts.map(methodContext => methodContext.contextValues).sort(function (a, b) {
+                if (a.name < b.name) {
```

Comment:
Sort functions are already implemented:
```typescript
methodContext.sort((a,b) => a.name.localeCompare(b.name))
```

Reply:
I already tried these but they are missing a check if the name property is set. 
Using the sort and localCompare methods, this is the error: 
![image](https://github.com/telekom/testerra/assets/138661340/375e6090-c52d-4eda-ad7f-0c5290d4f412)
@TobiasAdler gave me the hint to sort it that way, as he did in the threads view (https://github.com/telekom/testerra/blob/master/report-ng/app/src/components/threads/threads.ts#L140C23-L140C23)

Reply:
Well, then filter by valid data beforehand:
```typescript
methodContexts.filter(methodContext => methodContext.name?.trim().length > 0).sort(...)
```